### ISSUE TITLE
Mouse Lags and Parsing error

### ISSUE THREAD

**Issue by joshuader6** [rubberduck-vba/Rubberduck#1368]
Hello i just tried the new Alpha (hotfix1) 
First of all i noticed quiete a mouse lag when i open the vbe on an existing workbook (with a lot of code in it) 
After a few seconds the mouselag goes away as the paser falis to continue.
I cant seem to figure out where it falis. EDIT: I cant bring up the Parster errors window since none of the windows (Code inspections, source control, etc ) are showing. settings and about are working though...
![image](https://cloud.githubusercontent.com/assets/10895412/14776807/19c30486-0ac9-11e6-863e-502df8a747fd.png)

also if i open a new workbook the rubberduck status doesnt show the shiny tooltim wether the cursor is in a function or sub.
![image](https://cloud.githubusercontent.com/assets/10895412/14776850/6ac4e9f8-0ac9-11e6-9766-a1675207fe5b.png)
![image](https://cloud.githubusercontent.com/assets/10895412/14776856/7566d402-0ac9-11e6-97b1-28e43b5ca937.png)

aand lastly sometimes when i close excel i get this error message: 
[http://pastebin.com/fUD2UBjy](http://pastebin.com/fUD2UBjy)

I hope this helps you getting closer to a beta or even an stable release :O 
Keep up the great work :)

**Comment by retailcoder** [comment]
Thanks for the feedback!

That exception looks like an easy fix, we'd just need to check if the key is null before trying to remove it from the dictionary... now the real question is *why would that key be null in the first place*...

When the "status bar" is showing an `Error` status (I need to lookup the German translation!), and you click that error status label - it should bring up a "Parser Errors" search results toolwindow. If it doesn't, then there's no `SyntaxErrorException` and *something else* is causing an "error" state; we definitely need to enable logging there.

The command for bringing up the code inspections, and other toolwindows that would require a "Ready" state to work properly, are disabled (by design) when parser state isn't "Ready" (although the hotkey for them doesn't look at whether the command is enabled or not, so the hotkey would still bring it up lol).

As for the "shiny tooltip", it's only going to show the content of a `VB_Description` procedure attribute if the module parses correctly. You can edit that attribute via the *object browser*, by browsing to the member you want, right-click and select "properties" to display the *member options*:

![image](https://cloud.githubusercontent.com/assets/5751684/14790916/79abcc30-0ae1-11e6-93eb-42b6ad8b2d32.png)

The object browser will then show that description:

![image](https://cloud.githubusercontent.com/assets/5751684/14790926/8bb475bc-0ae1-11e6-8c20-d40d5bce7571.png)

And when you re-parse (Rubberduck will export the module to a temp file, parse the module attributes, discard the temp file, and then parse the code in the code pane and assign the collected attributes to their respective members), the status bar will show the "shiny tooltip" whenever the current selection is on the procedure with the description attribute you just entered.

**Comment by retailcoder** [comment]
The low-level mouse hook is to blame for the mouse lag - it was much worse (unbearable actually) before I introduced a bit of caching to "remember" what the "last selected declaration" was. But still *every time the selection changes in the VBE*, Rubberduck needs to locate *what's selected* - and then determine whether menu items should be enabled or not, e.g. "Refactor / Encapsulate Field" should only be enabled when a public field variable is selected; "Refactor / Extract Method" should only be enabled when the current selection is valid for that operation, etc.

The hooks are disabled when the VBE loses focus, and re-enabled when it gets it back.

There's definitely room for improvement there - the worst lags I've experienced was when dragging a toolwindow, and for some reason when unwinding the stack after an exception occurred (well, in debug mode - mouse lags for a few seconds in Visual Studio, and then everything goes back to normal).

Not sure what the fix would be though.

**Comment by joshuader6** [comment]
Okay I thought that the "siny tooltip" would be showing the actual selection context. So whether the selection is in a function or a sub or whatsoever. 

As for the parser error. No when I click the label nothing happens. Strangely i noticed that the locale in settings is set to English so it seems weird that the label is localized German. 

Sadly I can't post the entire document or even the code of it because it contains confidential information. 

Would it be possible to somehow catch the error and at least display an error message or where in the vba code it is happening? 

I could provide limited snippets but not very more. 

And concerning the error message happening on excel close: it happens completely random. Sometimes the error occurres sometimes it doesn't . But it seems to be more frequent when there was no saved workbook open. So just open excel and add some code to the empty workbook and close it right away.

**Comment by retailcoder** [comment]
We definitely need to make this more discoverable... when the status label in the Rubberduck commandbar says `Parse Error` (for `ParserState.Error` state), clicking it brings up a "Parser Errors" toolwindow:

![image](https://cloud.githubusercontent.com/assets/5751684/14793946/81890298-0aef-11e6-945f-fdfe3d33ba26.png)

You can double-click items in that list to navigate there (note: it won't refresh automatically).

When parser state is `Ready`, you *should* get context-sensitive information about what the current selection is:

![image](https://cloud.githubusercontent.com/assets/5751684/14794053/fcf93d80-0aef-11e6-92e4-ff089d36b3ae.png)

now, `ThisWorkbook` and other document-type modules can't be exported, so procedure attributes won't work, but if the member is in a standard module, class module, or a user form, re-parsing after adding a description attribute *should* give you that tooltip:

![image](https://cloud.githubusercontent.com/assets/5751684/14794186/a335f102-0af0-11e6-8ef6-73043172303c.png)

**Comment by joshuader6** [comment]
Alright i just checked: On a new workbook the ParserError window works. 
But when i tried to open the Document wich is causing trouble i encountert a new error: 
even bevore opening the VBE: 
![image](https://cloud.githubusercontent.com/assets/10895412/14794550/00c83fbc-0b25-11e6-9d14-dff79ad750cc.png)

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.Runtime.InteropServices.COMException (0x8001010A): Durch den Messagefilter wurde angezeigt, dass die Anwendung ausgelastet ist. (Ausnahme von HRESULT: 0x8001010A (RPC_E_SERVERCALL_RETRYLATER))
   at Microsoft.Vbe.Interop.VBE.get_ActiveVBProject()
   at Rubberduck.VBEditor.Extensions.VbeExtensions.HostApplication(VBE vbe) in c:\Users\Mathieu\Documents\GitHub\Rubberduck\Rubberduck.VBEEditor\Extensions\VbeExtensions.cs:line 45
   at Rubberduck.Inspections.ImplicitActiveSheetReferenceInspection.GetInspectionResults() in c:\Users\Mathieu\Documents\GitHub\Rubberduck\RetailCoder.VBE\Inspections\ImplicitActiveSheetReferenceInspection.cs:line 34
   at Rubberduck.Inspections.Rubberduck.Inspections.Inspector.<>c__DisplayClass4.<>c__DisplayClass6.<FindIssuesAsync>b__2() in c:\Users\Mathieu\Documents\GitHub\Rubberduck\RetailCoder.VBE\Inspections\Inspector.cs:line 69
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1078.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
Microsoft.VisualStudio.Tools.Office.Runtime
    Assembly Version: 10.0.0.0
    Win32 Version: 10.0.60301.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualStudio.Tools.Office.Runtime/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualStudio.Tools.Office.Runtime.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1075.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
mscorlib.resources
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_de_b77a5c561934e089/mscorlib.resources.dll
----------------------------------------
Microsoft.VisualStudio.Tools.Applications.Hosting
    Assembly Version: 10.0.0.0
    Win32 Version: 10.0.60301.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualStudio.Tools.Applications.Hosting/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualStudio.Tools.Applications.Hosting.dll
----------------------------------------
Microsoft.VisualStudio.Tools.Applications.ServerDocument
    Assembly Version: 10.0.0.0
    Win32 Version: 10.0.60301.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualStudio.Tools.Applications.ServerDocument/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualStudio.Tools.Applications.ServerDocument.dll
----------------------------------------
Microsoft.VisualStudio.Tools.Applications.Runtime
    Assembly Version: 10.0.0.0
    Win32 Version: 10.0.60301.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualStudio.Tools.Applications.Runtime/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualStudio.Tools.Applications.Runtime.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1078.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
Microsoft.VisualStudio.Tools.Office.Runtime.resources
    Assembly Version: 10.0.0.0
    Win32 Version: 10.0.60301.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualStudio.Tools.Office.Runtime.resources/v4.0_10.0.0.0_de_b03f5f7f11d50a3a/Microsoft.VisualStudio.Tools.Office.Runtime.resources.dll
----------------------------------------
System.Deployment
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1078.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Deployment/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Deployment.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1064.2 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Security
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1073.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Security/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Security.dll
----------------------------------------
System.Deployment.resources
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Deployment.resources/v4.0_4.0.0.0_de_b03f5f7f11d50a3a/System.Deployment.resources.dll
----------------------------------------
System.Xml.Linq
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml.Linq/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.Linq.dll
----------------------------------------
Microsoft.Office.Tools
    Assembly Version: 10.0.0.0
    Win32 Version: 10.0.60301.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.Office.Tools/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.Office.Tools.dll
----------------------------------------
Rubberduck
    Assembly Version: 2.0.0.2650
    Win32 Version: 2.0.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Rubberduck.DLL
----------------------------------------
Ninject
    Assembly Version: 3.2.0.0
    Win32 Version: 3.2.2.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Ninject.DLL
----------------------------------------
Ninject.Extensions.Factory
    Assembly Version: 3.2.0.0
    Win32 Version: 3.2.1.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Ninject.Extensions.Factory.DLL
----------------------------------------
Castle.Core
    Assembly Version: 3.2.0.0
    Win32 Version: 3.2.0.14
    CodeBase: file:///C:/ProgramData/Rubberduck/Castle.Core.DLL
----------------------------------------
Microsoft.Vbe.Interop
    Assembly Version: 15.0.0.0
    Win32 Version: 15.0.4561.1000
    CodeBase: file:///C:/Windows/assembly/GAC_MSIL/Microsoft.Vbe.Interop/15.0.0.0__71e9bce111e9429c/Microsoft.Vbe.Interop.dll
----------------------------------------
Rubberduck.Parsing
    Assembly Version: 2.0.5959.2646
    Win32 Version: 2.0.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Rubberduck.Parsing.DLL
----------------------------------------
Rubberduck.VBEditor
    Assembly Version: 1.0.5959.2645
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Rubberduck.VBEditor.DLL
----------------------------------------
Rubberduck.SourceControl
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Rubberduck.SourceControl.DLL
----------------------------------------
Rubberduck.SmartIndenter
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Rubberduck.SmartIndenter.DLL
----------------------------------------
PresentationCore
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1073.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_32/PresentationCore/v4.0_4.0.0.0__31bf3856ad364e35/PresentationCore.dll
----------------------------------------
WindowsBase
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1073.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/WindowsBase/v4.0_4.0.0.0__31bf3856ad364e35/WindowsBase.dll
----------------------------------------
PresentationFramework
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1073.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework/v4.0_4.0.0.0__31bf3856ad364e35/PresentationFramework.dll
----------------------------------------
System.Xaml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1073.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xaml/v4.0_4.0.0.0__b77a5c561934e089/System.Xaml.dll
----------------------------------------
office
    Assembly Version: 15.0.0.0
    Win32 Version: 15.0.4613.1000
    CodeBase: file:///C:/Windows/assembly/GAC_MSIL/office/15.0.0.0__71e9bce111e9429c/office.dll
----------------------------------------
System.Windows.Interactivity
    Assembly Version: 4.0.0.0
    Win32 Version: 2.0.20525.0
    CodeBase: file:///C:/ProgramData/Rubberduck/System.Windows.Interactivity.DLL
----------------------------------------
Ninject.Extensions.Conventions
    Assembly Version: 3.2.0.0
    Win32 Version: 3.2.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Ninject.Extensions.Conventions.DLL
----------------------------------------
Antlr4.Runtime.net45
    Assembly Version: 4.3.0.0
    Win32 Version: 4.3.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Antlr4.Runtime.net45.DLL
----------------------------------------
Ninject.Extensions.NamedScope
    Assembly Version: 3.2.0.0
    Win32 Version: 3.2.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Ninject.Extensions.NamedScope.DLL
----------------------------------------
DynamicProxyGenAssembly2
    Assembly Version: 0.0.0.0
    Win32 Version: 3.2.0.14
    CodeBase: file:///C:/ProgramData/Rubberduck/Castle.Core.dll
----------------------------------------
DynamicProxyGenAssembly2
    Assembly Version: 0.0.0.0
    Win32 Version: 3.2.0.14
    CodeBase: file:///C:/ProgramData/Rubberduck/Castle.Core.dll
----------------------------------------
WindowsFormsIntegration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/WindowsFormsIntegration/v4.0_4.0.0.0__31bf3856ad364e35/WindowsFormsIntegration.dll
----------------------------------------
Rubberduck.resources
    Assembly Version: 2.0.0.2650
    Win32 Version: 2.0.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/de/Rubberduck.resources.DLL
----------------------------------------
Infralution.Localization.Wpf
    Assembly Version: 3.3.0.0
    Win32 Version: 3.3.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/Infralution.Localization.Wpf.dll
----------------------------------------
PresentationFramework.Aero2
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework.Aero2/v4.0_4.0.0.0__31bf3856ad364e35/PresentationFramework.Aero2.dll
----------------------------------------
PresentationFramework.resources
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework.resources/v4.0_4.0.0.0_de_31bf3856ad364e35/PresentationFramework.resources.dll
----------------------------------------
Anonymously Hosted DynamicMethods Assembly
    Assembly Version: 0.0.0.0
    Win32 Version: 4.6.1078.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_32/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll
----------------------------------------
PresentationFramework.Aero
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework.Aero/v4.0_4.0.0.0__31bf3856ad364e35/PresentationFramework.Aero.dll
----------------------------------------
System.Windows.Forms.resources
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_de_b77a5c561934e089/System.Windows.Forms.resources.dll
----------------------------------------
NLog
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.0.0
    CodeBase: file:///C:/ProgramData/Rubberduck/NLog.DLL
----------------------------------------
System.ServiceModel
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.ServiceModel/v4.0_4.0.0.0__b77a5c561934e089/System.ServiceModel.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_32/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
System.resources
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.resources/v4.0_4.0.0.0_de_b77a5c561934e089/System.resources.dll
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------
PresentationFramework-SystemXmlLinq
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework-SystemXmlLinq/v4.0_4.0.0.0__b77a5c561934e089/PresentationFramework-SystemXmlLinq.dll
----------------------------------------
PresentationFramework-SystemXml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework-SystemXml/v4.0_4.0.0.0__b77a5c561934e089/PresentationFramework-SystemXml.dll
----------------------------------------
PresentationCore.resources
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/PresentationCore.resources/v4.0_4.0.0.0_de_31bf3856ad364e35/PresentationCore.resources.dll
----------------------------------------
UIAutomationTypes
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/UIAutomationTypes/v4.0_4.0.0.0__31bf3856ad364e35/UIAutomationTypes.dll
----------------------------------------
CustomMarshalers
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_32/CustomMarshalers/v4.0_4.0.0.0__b03f5f7f11d50a3a/CustomMarshalers.dll
----------------------------------------
stdole
    Assembly Version: 7.0.3300.0
    Win32 Version: 7.00.9466
    CodeBase: file:///C:/Windows/assembly/GAC/stdole/7.0.3300.0__b03f5f7f11d50a3a/stdole.dll
----------------------------------------
Microsoft.CSharp
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.CSharp/v4.0_4.0.0.0__b03f5f7f11d50a3a/Microsoft.CSharp.dll
----------------------------------------
PresentationFramework-SystemData
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework-SystemData/v4.0_4.0.0.0__b77a5c561934e089/PresentationFramework-SystemData.dll
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Dynamic
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Dynamic/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Dynamic.dll
----------------------------------------
System.Dynamic.DynamicAssembly
    Assembly Version: 0.0.0.0
    Win32 Version: 4.6.1038.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Dynamic/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Dynamic.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.



```

This is on a different machine than tomorrow though.

On this machine i also encountert the biggest mouse lag ever :D 
When i try to move the Parser error window on another (very code intensive) Workbook. 
The Mouse starts behaving very funky. 
The window will move sort of in slow motion and the mouse will only let go after roughly 10 seconds :D

**Comment by joshuader6** [comment]
so it took me like 2 minutes to move the window there for the screenshot xD
But now this is happening: 
![image](https://cloud.githubusercontent.com/assets/10895412/14794813/fe94c0d4-0b25-11e6-8816-6b203bdc7f39.png)

**Comment by BobTB** [comment]
Hi, 
These are actually exactly the same problems I have while using the latest alpha in Access,  if it has any significant amount of code.  Parser errors  tool window does not open up,  huge lag in vbe editor,  random crashes on closing access

**Comment by Vogel612** [comment]
@joshuader6 This error is a manifestation of the application being overstrained. The mouse lag increase definitely hints at lower system specs. Would you mind sharing your Hardware setup?

---

Dieser Fehler deutet auf eine Überlastung der Anwendung hin. Dass die Maus stärker hängt deutet definitiv auf niedrigere System-Power hin. Könntest du vielleicht deine Hardware-Konfiguration auflisten?

**Comment by Vogel612** [comment]
@rubberduck-vba/dev Error message roughly translated from German:

    COMException: The Messagefilter indicated that the Application is busy. (Exception HRESULT: 0x8001010A [...]

**Comment by retailcoder** [comment]
@Vogel612 nein - or I have sub-par hardware as well! There's definitely something with the mouse hook not handling things as intended.

**Comment by joshuader6** [comment]
neither of the machines have low specs i would say ^^

Machine 1 (this morning): HP laptop worksation with some intel xenon processor (8 threads) , Nvidia Quadro K2000 , 32gb ram 

Machine2 : Lenovo Y580 i7 3610QM , 16gb ram , Nvidia gtx660m, Samsung 840evo ssd as main drive

**Comment by joshuader6** [comment]
To clean this up a bit i've moved the large Exception dumps to pastebin :)

**Comment by BobTB** [comment]
Just to chime in, my machine has 32GB RAM, 4-Core i7, 512GB SSD as main drive, and still completely stalls on any larger code being parsed.

**Comment by retailcoder** [comment]
@BobTB makes me wonder if background-thread parsing was a good idea at all... should we revert to 1.x-style "Please wait" blocking dialog? (I'd hate to :unamused: )

**Comment by BobTB** [comment]
I don't know much about multi threading/ background thread parsing and stuff, but if it could do what it does in some very low priority process, it would probably be ok.

Otherwise:

1. Its better to wait it out, and be done with it, to know that it is done
2. It then does not slow down the whole PC while nothing visible is happening

**Comment by joshuader6** [comment]
Seems like this is Fixed as of Commit: 3b659333e050038b44e7754340a2a6b8846bdbdc
